code/fireball/fireballs.cpp

```diff
@@ -1093,3 +1067,14 @@ void fireball_render(object* obj, model_draw_list *scene)
 			Int3();
 	}
 }
+
+int fireball_get_count()
+{
+	int count = static_cast<int>(Fireballs.size() - Unused_fireball_indices.size());
+
+	if (count < 0) {
```

Comment:
This is a surprisingly simple way of doing this. Maybe add a comment in this function to explain why this works to make understanding this a bit easier.

Also, be careful with your casting range checks. The way this currently works means that this will not be hit in the cases you want. Since both `size` calls return a `size_t`, you will get an underflow if `Unused_fireball_indices` is larger than `Fireballs`. If you do the casts individually then the result will be as you expect it.

Also, is this a case you would expect? I can't think of something where this could happen without a programming error in which case an assertion would be sufficient.

Reply:
I am not expecting it, I just got paranoid. So you're right, I should make it an assert.